### PR TITLE
Openstack telemetry aodh support

### PR DESCRIPTION
**Pull request by zhangjianweibj**
telemetry_aodh support

### CODE REVIEW

**Comment by zhangjianweibj** [comment]
fix https://github.com/ContainX/openstack4j/issues/896

**Comment by zhangjianweibj** [comment]
the new feature documented in http://docs.openstack.org/developer/aodh/, which is designed to scale telemetry alarm service,but this new feature is not supported by openstack4j .

**Comment by auhlig** [comment]
Hi @zhangjianweibj ,
Thanks for filing the issue and creating a PR.
Before we can take this further, I'd like to ask you to add some tests.
Feel free to reach out in case you need help.

**Comment by zhangjianweibj** [comment]
hi @auhlig ,telemetry aodh test cases already commited

File: core-test/src/main/java/org/openstack4j/api/telemetry/CapabilitiesV3Test.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Override
+    protected Service service() {
+        return Service.TELEMETRY;
+    }
+
+    @Test
+    public void getSampleTest() throws IOException {
+        respondWith(JSON_CAPABILITIES);
+        Capabilities capabilities = osv3().telemetry().capabilities().get();
```
Comment (auhlig) [inline core-test/src/main/java/org/openstack4j/api/telemetry/CapabilitiesV3Test.java]: Shouldn't this be like `.get(<id>)` ? Else maybe `.list()` is be more consistent, or?

Comment (auhlig) [inline core-test/src/main/java/org/openstack4j/api/telemetry/CapabilitiesV3Test.java]: Maybe also check at at least one attribute.

File: core/src/main/java/org/openstack4j/openstack/telemetry/internal/BaseTelemetryAodhServices.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.List;
+
+public class BaseTelemetryAodhServices extends BaseOpenStackService {
+
+	protected BaseTelemetryAodhServices() {
+		super(ServiceType.TELEMETRY_AODH, EndpointFunction.instance);
+	}
+	
+
+	private static class EndpointFunction implements Function<String, String> {
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/telemetry/internal/BaseTelemetryAodhServices.java]: Could you briefly describe why this is necessary?
Usually OpenStack services get the endpoint from the service catalog.

**Comment by vinodborole** [comment]
i agree @auhlig

File: core/src/main/java/org/openstack4j/openstack/telemetry/internal/BaseTelemetryAodhServices.java (hunk 2 above)
Comment (zhangjianweibj) [inline core/src/main/java/org/openstack4j/openstack/telemetry/internal/BaseTelemetryAodhServices.java]: yeah,OpenStack services can get the endpoint from the service catalog.but openstack4j is a tool for  developers.users can get service information  from osclient.BaseTelemetryAodhServices is responsible for the related work.

Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/telemetry/internal/BaseTelemetryAodhServices.java]: Understood. I don't really like doing that here. Maybe we could the include that in the path constant instead. Does that make sense?

Comment (zhangjianweibj) [inline core/src/main/java/org/openstack4j/openstack/telemetry/internal/BaseTelemetryAodhServices.java]: thank for your suggestion,path constant maybe a good idea at present.but,telemetry service endpoint may be changed with version information or not in the future. EndpointFunction solution can work with that situation.  another solution we can do like
image

what is your idea,auhlig?

Comment (zhangjianweibj) [inline core/src/main/java/org/openstack4j/openstack/telemetry/internal/BaseTelemetryAodhServices.java]: image

**Pull request merged**